### PR TITLE
Feature/observer

### PR DESCRIPTION
**Pull request by straleyb**

### CODE REVIEW

File: app/assets/javascripts/ordered_creator.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
-      hidden_index_selectors[index].value = index;
-    });
+  selector.nestable({ maxDepth: 0, rootClass: 'ordered-field-container' });
+  selector.on('change', function (event) {
+    reindexNestedOrderedField();
+  });
+}
+
+function reindexNestedOrderedField(mutationsList) {
+  hidden_index_selectors = $($('.ordered-field-container')).find(".index");
```
Comment (revgum) [inline app/assets/javascripts/ordered_creator.js]: clean up the jquery syntax?

Comment (revgum) [reply app/assets/javascripts/ordered_creator.js]:
```
$('.ordered-field-container').find(".index")

OR 

$('.ordered-field-container .index')
```

File: app/inputs/multi_value_ordered_input.rb [hunk 2]
```diff
@@ -29,7 +29,7 @@ def inner_wrapper(value, index)
   end
 
   def outer_wrapper
-    " <ul class='listing draggable-order dd-list' data-object-name='#{object_name}'>\n
+    " <ul id='ordered-creator' class='listing draggable-order dd-list' data-object-name='#{object_name}'>\n
```
Comment (revgum) [inline app/inputs/multi_value_ordered_input.rb]: this shouldn't have an ID since it is intended to be a general reusable input

Comment (straleyb) [reply app/inputs/multi_value_ordered_input.rb]: 👍

File: app/assets/javascripts/ordered_creator.js (hunk 1 above)
Comment (straleyb) [reply app/assets/javascripts/ordered_creator.js]: 👍

File: app/assets/javascripts/ordered_creator.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ function orderedDragAndDrop(selector) {
 }
 
 function reindexNestedOrderedField(mutationsList) {
-  hidden_index_selectors = $($('.ordered-field-container')).find(".index");
+  hidden_index_selectors = $($(".ordered-field-container .index"));
```
Comment (revgum) [inline app/assets/javascripts/ordered_creator.js]: This is incorrect, you don't need nested jQuery here..

**Pull request merged**